src/components/Newsletter/Mailchimp.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import jsonp from "jsonp";
+import classnames from "classnames";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import styles from "./styles.module.css";
+
+// Mailchimp returns HTML in it's response so we return out own messages. We are doing string comparison here which is a bit nasty. Especially if the Mailchimp language changes.
+const handleResponse = (err, data) => {
+  if (data && data.msg.includes("already subscribed")) {
+    return {
+      message: "This email address is already subscribed to our list.",
+      result: "error",
+    };
+  } else if (err || data.result === "error") {
+    return {
+      message: "Something went wrong, please try again.",
+      result: "error",
+    };
+  } else {
+    return {
+      message: "Thank you for subscribing.",
+      result: data.result,
+    };
+  }
+};
+
+const ResponseMessage = ({ response }) => (
+  <p style={{ color: response.result === "error" ? "red" : "green" }}>
+    {response.message}
+  </p>
+);
+
+const Mailchimp = ({ action }) => {
+  const [response, setResponse] = useState(null);
+
+  const handleSubmit = values => {
+    // Unset reponse data
+    setResponse(null);
+
+    const params = Object.entries(values)
+      .map(([key, val]) => `${key.toUpperCase()}=${val}`)
+      .join("&");
+    const path = `${action}&${params}`;
+
+    return new Promise((resolve) => {
+      jsonp(path, { param: "c" }, (err, data) => {
```

Comment:
What does the `jsonp` call do here? Is it required by Mailchimp?

Reply:
@nimobeeren - I took inspiration from this [package](https://github.com/gndx/react-mailchimp-form/blob/master/src/index.js). I was going to use the package but it didn't quite do what I wanted. I'll check there a no security concerns etc using this method....if not it makes sense to keep it this way. I'll let you know

Reply:
`jsonp` is the format Mailchimp expects and this seems fine. Ideally we'd use the API but for the sake of simplicity I'm happy with this solution.